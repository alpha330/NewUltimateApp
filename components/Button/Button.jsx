/** @jsxImportSource @emotion/react */
import { css } from "@emotion/react";
import { useTheme } from "@emotion/react";

const Button = ({type="button",children,loading,disabled,handleClick=()=>{}}) => {
  const theme = useTheme()
  const btnStyle = css`
    width:100%;
    padding:10px;
    background-color:${theme.color.primary}
  `;
  console.log("button",loading)
  return(
    <>
    {disabled &&
    <button onClick={handleClick} disabled type={type} css={btnStyle} >
    {loading && "loading"}
    {!loading && children}
    </button>
    }
    {!disabled &&
    <button onClick={handleClick} type={type} css={btnStyle} >
    {loading && "loading"}
    {!loading && children}
    </button>
    }
    </>
  )
}

export default Button
