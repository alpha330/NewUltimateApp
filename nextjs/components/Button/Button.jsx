/** @jsxImportSource @emotion/react */
import { css } from "@emotion/react";
import { useTheme } from "@emotion/react";

const Button = ({type="button",children,loading}) => {
  const theme = useTheme()
  const btnStyle = css`
    width:100%;
    padding:10px;
    background-color:${theme.color.primary}
  `;
  console.log("button",loading)
  return(
    <button type={type} css={btnStyle}>
      {loading && "loading"}
      {!loading && children}
      </button>
  )
}

export default Button
