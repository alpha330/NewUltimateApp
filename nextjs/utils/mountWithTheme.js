import { ThemeProvider } from "@emotion/react";
import theme from "../configs/themes";
import renderer from 'react-test-renderer';
import React from "react";

const mountWithTheme = (children) => {
  return renderer.create(
    <ThemeProvider theme={theme}>
          {children}
    </ThemeProvider>
  )

}

export default mountWithTheme
