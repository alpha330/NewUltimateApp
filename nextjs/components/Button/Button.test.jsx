import Button from "./Button"
import mountWithTheme from "../../utils/mountWithTheme";
import React from "react";

describe ('Button', () => {
  test(`Render Correctly`,()=>{
    const component = mountWithTheme(<Button>Sample Button</Button>)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
