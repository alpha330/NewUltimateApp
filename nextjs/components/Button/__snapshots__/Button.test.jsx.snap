// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Render Correctly 1`] = `null`;

exports[`Button show disabled state 1`] = `null`;

exports[`Button show loading state 1`] = `null`;
